oecophylla/taxonomy/taxonomy.rule

```diff
@@ -200,19 +201,124 @@ rule taxonomy_kraken_combine_profiles:
+        with tempfile.TemporaryDirectory(dir=find_local_scratch(TMP_DIR_ROOT)) as temp_dir:
+            shell("""
+                  set +u; {params.env}; set -u
+
+                  # get stem file path
+                  stem={output.report}
+                  stem=${{stem%.report.txt}}
+
+                  # run Centrifuge to align reads against reference genomes
+                  centrifuge \
+                    -1 {input.forward} \
+                    -2 {input.reverse} \
+                    -x {params.db} \
+                    -p {threads} \
+                    -S {temp_dir}/map.tmp \
+                    --report-file {output.profile} \
+                    2>> {log} 1>&2
+
+                  # generate Kraken-style hierarchical report
+                  centrifuge-kreport {temp_dir}/map.tmp \
+                    -x {params.db} \
+                    1> {output.report} \
+                    2>> {log}
+
+                  # keep mapping file
+                  if [[ "{params.map}" == "True" ]]
+                  then
+                    gzip -c {temp_dir}/map.tmp > $stem.map.txt.gz
+                  fi
+
+                  # run Bracken to re-estimate abundance at given rank
+                  if [[ ! -z {params.levels} ]]
+                  then
+                    IFS=',' read -r -a levels <<< "{params.levels}"
+                    for level in "${{levels[@]}}"
+                    do
+                      est_abundance.py -i {output.report} \
+                        -k {params.kmers} \
+                        -t 10 \
+                        -l $(echo $level | head -c 1 | tr a-z A-Z) \
+                        -o $stem.redist.$level.txt \
+                        2>> {log} 1>&2
+                      rm $stem.report_bracken.txt
+                    done
+                  fi
+                  """)
+
+
+rule taxonomy_centrifuge_combine_profiles:
+    """
+    Combines per-sample Centrifuge/Bracken output tables into single OTU
+    tables.
+    """
+    input:
+        expand(taxonomy_dir + "{sample}/centrifuge/{sample}.profile.txt",
+               sample=samples)
+    output:
+        taxonomy_dir + "centrifuge/combined_profile.biom"
+        # extra output files:
+        # combined_redist.{level}.biom foreach {levels}
+    params:
+        levels = config['params']['centrifuge']['levels']
+    log:
+        taxonomy_dir + "logs/taxonomy_centrifuge_combine_profiles.log"
+    benchmark:
+        "benchmarks/taxonomy/taxonomy_centrifuge_combine_profiles.txt"
+    run:
+        pandas2biom(output[0], combine_centrifuge(zip(samples, input)))
+        for level in params['levels'].split(','):
+            redists = ['%s.redist.%s.txt' % (x[:-12], level)
+                       for x in input]
+            pandas2biom('%s_redist.%s.biom' % (output[0][:-13], level),
+                        combine_bracken(zip(samples, redists)))
+
+
+rule centrifuge:
+    input:
+        taxonomy_dir + "centrifuge/combined_profile.biom"
```

Comment:
can also change to `rules.taxonomy_centrifuge_combine_profiles.output` 

This way if you update that output don't need to catch the change in import!

Reply:
Jon I didn't quite get your rationale. Can you elaborate little bit?